storage/src/linear/filebacked.rs

```diff
@@ -64,6 +64,19 @@ impl std::fmt::Debug for FileBacked {
 }
 
 impl FileBacked {
+    /// Acquire an advisory lock on the underlying file to prevent multiple processes
+    /// from accessing it simultaneously
+    pub fn lock(&self) -> Result<(), FileIoError> {
+        self.fd.try_lock().map_err(|e| {
```

Comment:
is `e` not already an `std::io::Error`?

Reply:
Nope. It's one of these: https://doc.rust-lang.org/beta/std/fs/enum.TryLockError.html

Reply:
yeah, [it is](https://doc.rust-lang.org/stable/std/fs/struct.File.html#method.lock) so wrapping it in another io error is unnecessary.